### PR TITLE
Update README.md : add a FAQ about memory release

### PR DESCRIPTION
**Pull request by dota17**
Refer to #1924 .

I think it is a very import case that we should add documents about it in README.md.
For me, it is helpful if I can find the case in README.md instead of searching issues.

Also, if we find a perfect solution in #1924 or other place, please update the README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1381,6 +1381,18 @@ This library will not support comments in the future. If you wish to use comment
 
 By default, the library does not preserve the **insertion order of object elements**. This is standards-compliant, as the [JSON standard](https://tools.ietf.org/html/rfc8259.html) defines objects as "an unordered collection of zero or more name/value pairs". If you do want to preserve the insertion order, you can specialize the object type with containers like [`tsl::ordered_map`](https://github.com/Tessil/ordered-map) ([integration](https://github.com/nlohmann/json/issues/546#issuecomment-304447518)) or [`nlohmann::fifo_map`](https://github.com/nlohmann/fifo_map) ([integration](https://github.com/nlohmann/json/issues/485#issuecomment-333652309)).
 
+### Memory Release
+
+We checked with Valgrind and the Clang Sanitizers that there are no memory leaks. 
+
+If you find that a prasering program with this library does not release memory, please consider these case and it maybe unrelated to this library.
```
Comment (nlohmann) [inline README.md]: Typo: "these case" -> "these cases".

File: README.md [hunk 2]
```diff
@@ -1381,6 +1381,18 @@ This library will not support comments in the future. If you wish to use comment
 
 By default, the library does not preserve the **insertion order of object elements**. This is standards-compliant, as the [JSON standard](https://tools.ietf.org/html/rfc8259.html) defines objects as "an unordered collection of zero or more name/value pairs". If you do want to preserve the insertion order, you can specialize the object type with containers like [`tsl::ordered_map`](https://github.com/Tessil/ordered-map) ([integration](https://github.com/nlohmann/json/issues/546#issuecomment-304447518)) or [`nlohmann::fifo_map`](https://github.com/nlohmann/fifo_map) ([integration](https://github.com/nlohmann/json/issues/485#issuecomment-333652309)).
 
+### Memory Release
+
+We checked with Valgrind and the Clang Sanitizers that there are no memory leaks. 
+
+If you find that a prasering program with this library does not release memory, please consider these case and it maybe unrelated to this library. 
+
+- Case 1: Your program is compiled with glibc.
```
Comment (nlohmann) [inline README.md]: If you only describe one case, there should not be a list starting "Case 1".

File: README.md [hunk 3]
```diff
@@ -1381,6 +1381,18 @@ This library will not support comments in the future. If you wish to use comment
 
 By default, the library does not preserve the **insertion order of object elements**. This is standards-compliant, as the [JSON standard](https://tools.ietf.org/html/rfc8259.html) defines objects as "an unordered collection of zero or more name/value pairs". If you do want to preserve the insertion order, you can specialize the object type with containers like [`tsl::ordered_map`](https://github.com/Tessil/ordered-map) ([integration](https://github.com/nlohmann/json/issues/546#issuecomment-304447518)) or [`nlohmann::fifo_map`](https://github.com/nlohmann/fifo_map) ([integration](https://github.com/nlohmann/json/issues/485#issuecomment-333652309)).
 
+### Memory Release
+
+We checked with Valgrind and the Clang Sanitizers that there are no memory leaks.
```
Comment (nlohmann) [inline README.md]: "the Clang Sanitizers" -> "the Address Sanitizer (ASAN)"

File: README.md [hunk 4]
```diff
@@ -1381,6 +1381,18 @@ This library will not support comments in the future. If you wish to use comment
...
 
+### Memory Release
+
+We checked with Valgrind and the Clang Sanitizers that there are no memory leaks. 
+
+If you find that a prasering program with this library does not release memory, please consider these case and it maybe unrelated to this library. 
+
+- Case 1: Your program is compiled with glibc. 
+
+	> There is a tunable threshold that glibc uses to decide whether to actually return memory to the system or whether to cache it for later reuse. If in your program you make lots of small allocations and those small allocations are not a contiguous block and are presumably below the threshold, then they will not get returned to the OS.
```
Comment (nlohmann) [inline README.md]: Rather add the paragraph as is. The quotation is confusing.

File: README.md [hunk 5]
```diff
@@ -1381,6 +1381,18 @@ This library will not support comments in the future. If you wish to use comment
...
+Here is a related issue [#1924](https://github.com/nlohmann/json/issues/1924).
+Also you can check your program under windows to ensure whether there are memory leaks.
```
Comment (nlohmann) [inline README.md]: I would remove the sentence "Also you can check your program under windows to ensure whether there are memory leaks.".

File: README.md [hunk 6]
```diff
@@ -1381,6 +1381,15 @@ This library will not support comments in the future. If you wish to use comment
 
 By default, the library does not preserve the **insertion order of object elements**. This is standards-compliant, as the [JSON standard](https://tools.ietf.org/html/rfc8259.html) defines objects as "an unordered collection of zero or more name/value pairs". If you do want to preserve the insertion order, you can specialize the object type with containers like [`tsl::ordered_map`](https://github.com/Tessil/ordered-map) ([integration](https://github.com/nlohmann/json/issues/546#issuecomment-304447518)) or [`nlohmann::fifo_map`](https://github.com/nlohmann/fifo_map) ([integration](https://github.com/nlohmann/json/issues/485#issuecomment-333652309)).
 
+### Memory Release
+
+We checked with Valgrind and the Address Sanitizer (ASAN) that there are no memory leaks. 
+
+If you find that a prasering program with this library does not release memory, please consider the following case and it maybe unrelated to this library.
```
Comment (nlohmann) [inline README.md]: What is `prasering`?

Comment (dota17) [reply README.md]: Sorry, just typo.

**Comment by dota17** [comment]
Done.

**Comment by nlohmann** [comment]
Thanks!

**Comment by nlohmann** [comment]
* * *

### 🔖 Release item

This issue/PR will be part of the next release of the library. This template helps preparing the release notes.

#### Type

- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] ⚡️ Improvement
- [x] 🔨 Further Change
- [ ] 🔥 Deprecated function

#### Description

- Add note to memory release #1933 #1924

* * *

**Pull request merged**